locust/argument_parser.py

```diff
@@ -271,14 +252,27 @@ def wait_for_reply():
         sys.stderr.write(f"Got error from master: {msg.data['error']}\n")
         sys.exit(1)
 
-    filename = msg.data["filename"]
-    with open(os.path.join(tempfile.gettempdir(), filename), "w", encoding="utf-8") as locustfile:
-        locustfile.write(msg.data["contents"])
+    tempclient.close()
+    return msg.data
 
-    atexit.register(exit_handler, locustfile.name)
 
-    tempclient.close()
-    return locustfile.name
+def parse_locustfile_paths_from_master(master_host, master_port):
+    locustfiles = download_locustfile_from_master(master_host, master_port).get("locustfiles", [])
+
+    def create_locustfile(file_to_create):
+        filename = file_to_create["filename"]
+        file_contents = file_to_create["contents"]
+
+        with open(os.path.join(tempfile.gettempdir(), filename), "w", encoding="utf-8") as locustfile:
+            locustfile.write(file_contents)
+
+        return locustfile.name
+
+    locustfiles = [
+        create_locustfile(locustfile) if "contents" in locustfile else locustfile for locustfile in locustfiles
```

Comment:
I think this would be easier to read without the internal function and list comprehension? Also, parse_locustfile_paths_from_master is both long and not very descriptive, because it does all sorts of stuff not related to parsing of paths (and what "parsing a path" means is also unclear :)

Reply:
Yeah I agree maybe this was a bit over-engineered trying to make list comprehension work haha! I broke it down into a for loop now instead.

Naming is tough, perhaps this function is doing too much and that's why it's difficult to name. I went with `parse_locustfiles_from_master` to mimic the naming of the other function `parse_locustfile_paths`. But I agree, parsing could mean anything.

My one issue with breaking this function down more would be that then I feel like the argument parser is doing too much, so there's not much winning. But another option could be something like this:
```
locustfile_sources = download_locustfile_from_master(options.master_host, options.master_port).get("locustfiles", [])
locustfile_list = parse_locustfiles_from_master(locustfile_sources)
return parse_locustfile_paths(locustfile_list)
```
or even have a shared function call like this:
```
if options.locustfile == "-":
    if not options.worker:
        sys.stderr.write(
            "locustfile was set to '-' (meaning to download from master) but --worker was not specified.\n"
        )
        sys.exit(1)
    # having this in argument_parser module is a bit weird, but it needs to be done early
    locustfile_sources = download_locustfile_from_master(options.master_host, options.master_port).get("locustfiles", [])
    locustfile_list = parse_locustfiles_from_master(locustfile_sources)
else:
    locustfile_list = [f.strip() for f in options.locustfile.split(",")]
parsed_paths = parse_locustfile_paths(locustfile_list)

if not parsed_paths:
    note_about_file_endings = ""
    user_friendly_locustfile_name = options.locustfile

    if not options.locustfile.endswith(".py"):
        note_about_file_endings = "Ensure your locustfile ends with '.py' or is a directory with parsed_paths. "
    sys.stderr.write(
        f"Could not find '{user_friendly_locustfile_name}'. {note_about_file_endings}See --help for available options.\n"
    )
    sys.exit(1)

return parsed_paths
```

Reply:
(I pushed the 2nd version in case you don't want to read all that, after thinking about it, I think it's easier to read)

Reply:
Looks nice now. Maybe **download**_locustfiles_from_master is good? and a type hint to indicate that it returns a list of strings